src/components/Modal/ModalHeader.js

```diff
@@ -1,48 +1,47 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { sparkComponentClassName } from '../../util'
+import classNames from 'classnames'
+import { sparkComponentClassName, sparkObjectClassName } from '../../util'
+import ModalHeading from './ModalHeading'
 
 class ModalHeader extends React.Component {
-  static defaultProps = {};
+  static defaultProps = {
+    onClose: () => {
+      console.log('closed')
+    }
+  };
 
   static propTypes = {
-    modalName: PropTypes.string.isRequired,
-    close: PropTypes.func,
+    onClose: PropTypes.func,
     id: PropTypes.string.isRequired,
-    text: PropTypes.string
+    title: PropTypes.string,
+    hasCloseButton: PropTypes.bool
   };
 
   get className() {
     const baseClass = sparkComponentClassName('Modal', 'header')
+    const stackItemClass = sparkObjectClassName('Stack', 'item')
 
-    return [baseClass].join('')
+    return classNames(baseClass, stackItemClass)
   }
 
   render = () => {
-    const { close, modalName, id, text } = this.props
+    const { onClose, id, title, hasCloseButton } = this.props
     return (
-      <div className={this.className}>
+      <header className={this.className}>
         {/* //todo: use react spark button/ */}
-        <h2 className='sprk-c-Modal__heading sprk-b-TypeDisplayFive' id={id}>
-          {text}
-        </h2>
-        <button
-          className='sprk-c-Modal__icon'
-          data-sprk-modal-cancel={modalName}
-          aria-label='Close Modal'
-          onClick={() => close()}
-        >
-          Close
-          <svg
-            className='sprk-c-Icon sprk-c-Icon--l sprk-c-Icon--current-color'
-            viewBox='0 0 64 64'
-            aria-hidden='true'
-            focusable='false'
-          >
-            <use xlinkHref='#close-circle' />
-          </svg>
-        </button>
-      </div>
+        <ModalHeading id={id}>
+          {title}
+        </ModalHeading>
+        {
+          hasCloseButton &&
+          <button onClick={onClose} className='sprk-c-Modal__icon' data-sprk-modal-cancel={id} type='button' aria-label='Close Modal'>
+            <svg className='sprk-c-Icon sprk-c-Icon--l sprk-c-Icon--current-color' viewBox='0 0 64 64' aria-hidden='true' focusable='false'>
```

Comment:
there is an Icon component now that we should use here
https://github.com/qloan/react-spark/blob/feat/redesign-phase1/src/components/Icon/Icon.js